### DISCUSSION TITLE
GPG encrypt secrets for self hosted instance

### DISCUSSION THREAD

**Discussion by FlxPeters** [renovatebot/renovate#18079] [Request Help] [answered]
Hey fellow community,
I try to encrypt some secrets for renovate on my self hosted instance. The docs are pretty clear how to create the keys, but not how to encrypt secret values.

**Do I miss something or what's the recommended way of encrypting secrets for an self hosted instance?**

I have some ideas like a simple bash script or self hosting the public encryption site. How do you solve this?

Edit: there is an example with terraform: https://github.com/bendrucker/terraform-gpg-renovate-encrypted-secret

**Comment by rarkins** [comment, marked as answer]
See https://docs.renovatebot.com/getting-started/private-packages/#encrypting-secrets

**Reply by FlxPeters** [reply]
Thanks, I missed the part of copying just the HTML page to encrypt my own secrets. I expected something like extra docker container, but this is obviously easier.

I have another question to encrypted secrets : how does the organization apply to Gitlab? Is it the group of an repository or can I use something more generic?

**Reply by rarkins** [reply]
Here is the logic if you scope to an org: https://github.com/renovatebot/renovate/blob/2abd4db5a235920b97c26711cb9643b2c8465be2/lib/config/decrypt.ts#L114-L128

In GitLab that maps cleaning to a top-level Group, possibly a subgroup would work too but I haven't tested it. Not sure what you mean by "more generic" but the `/` which is appended means it can't be a simple "prefix" of a group name.

**Reply by rarkins** [reply]
Remember if you're self=hosting you can also provision secrets on the server side config and you're not constrained to scopes

**Reply by FlxPeters** [reply]
Thanks for the hint. I use server side secrets for the moment but also want to understand how it works.
